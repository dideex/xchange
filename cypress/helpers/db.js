const mongoose = require('mongoose')
const mongoConfig = require('./config.json').mongodb

module.exports = mongoose.connect(mongoConfig, {useNewUrlParser: true})
