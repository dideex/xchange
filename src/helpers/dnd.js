import React from 'react'
import {DragDropContext} from 'react-dnd'
import TestBackend from 'react-dnd-test-backend'

export function wrapInTestContext(DecoratedComponent) {
  const TestStub = props => <DecoratedComponent {...props} />
  return DragDropContext(TestBackend)(TestStub)
}
