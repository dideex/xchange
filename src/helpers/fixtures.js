export const fakeCurrnecy = [
  {
    order: 1,
    _id: 1,
    icon: 'eth',
    id: 'eth',
    name: 'tether',
    label: 'tether label',
    source: 'tether source wallet'
  },
  {
    order: 2,
    _id: 2,
    icon: 'bitcoin',
    id: 'bitcoin',
    name: 'lisk',
    label: 'lisk label',
    source: 'lisk source wallet'
  },
  {
    order: 3,
    _id: 3,
    icon: 'rur',
    id: 'rur',
    name: 'litecoin',
    label: 'litecoin label',
    source: 'litecoin source wallet'
  },
]


export const fakeData = {
  id: 'Test id',
  inputValue: '1000',
  toWallet: 'Test wallet',
  currencyInputLabel: 'Test input label',
  currencyOutputLabel: 'Test output label',
  fromWallet: 'Tests fromWallet',
  outputValue: '2000',
  email: 'Tests email',
  username: 'Tests username',
  paymentStatus: 1,
  loading: false,
  wallets: {
    eth: 'eth wallet',
    bitcoint: 'bitcoint wallet',
    rur: 'rur wallet',
  },
  login: 'Test login',
  updatePaymentStatus() {},
}

export const fakeUser = {
  username: 'Fake username',
  password: 'Fake password',
  email: 'fake@email.com',
  phone: '89145678901',
  fio: 'Fake fio',
  token: 'Fake token',
}

export const fakeBitcoinAddress = '3JQSigWTCHyBLRD979JWgEtWP5YiiFwcQB '
export const fakeCreditAddress = '1234 4321 5678 8765'

export const fakeWallets = {
  eth: 'Fake eth wallet',
  bitcoin: 'Fake bitcoin wallet',
  rur: 'Fake rur wallet'
}