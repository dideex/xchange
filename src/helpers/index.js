export {fakeCurrnecy} from './fixtures'

export const delay = (fn = () => {}, timeout = 0) =>
  new Promise(res =>
    setTimeout(() => {
      fn()
      res()
    }, timeout),
  )
