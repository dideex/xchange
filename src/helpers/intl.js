import React from 'react'
import {mount, shallow} from 'enzyme'
import {IntlProvider, intlShape} from 'react-intl'

import messages from '../locale/en.json'
const intlProvider = new IntlProvider({locale: 'en', messages}, {})
const {intl} = intlProvider.getChildContext()

function nodeWithIntlProp(node) {
  return React.cloneElement(node, {intl})
}

export function mountWithIntl(
  node,
  {context, childContextTypes, ...additionalOptions} = {},
) {
  return mount(nodeWithIntlProp(node), {
    context: Object.assign({}, context, {intl}),
    childContextTypes: Object.assign({}, {intl: intlShape}, childContextTypes),
    ...additionalOptions,
  })
}
