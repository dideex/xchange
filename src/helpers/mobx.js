import React from 'react'
import {Provider} from 'mobx-react'
import {lastOperations, cash, user} from '../store'

jest.mock('../components/Api', () => ({
  post: () => Promise.resolve({data: {}}),
  get: () => Promise.resolve({data: {}}),
  errorEmitter: jest.fn(data => fn => data(fn)),
}))

// jest.mock('../components/common', () => ({
//   noty: () => {},
//   Loading: () => <div>Loading</div>,
// }))

export const MobxProvider = ({children, ...stores}) => {
  return (
    <Provider
      lastOperationsStore={lastOperations}
      cashStore={cash}
      userStore={user}
      {...stores}
    >
      {children}
    </Provider>
  )
}
