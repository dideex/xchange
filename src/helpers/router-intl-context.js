import {mount} from 'enzyme'
import {BrowserRouter} from 'react-router-dom'
import {IntlProvider, intlShape} from 'react-intl'
import {shape} from 'prop-types'

import messages from '../locale/en.json'

export const testId = 'test id'

// Instantiate router context
const router = {
  history: {...new BrowserRouter().history},
  route: {
    location: {
      path: '/location',
      href: '/location',
    },
    match: {
      path: '/location',
      params: {
        id: testId,
      },
    },
  },
}

// const createContext = () => ({
//   context: {router},
//   childContextTypes: {router: shape({})},
// })

const intlProvider = new IntlProvider({locale: 'en', messages}, {})
const {intl} = intlProvider.getChildContext()

const createMixedContext = ({
  router,
  context,
  childContextTypes,
  ...additionalOptions
} = {}) => ({
  context: Object.assign({}, {router}, context, {intl}),
  childContextTypes: Object.assign(
    {},
    {router: shape({})},
    {intl: intlShape},
    childContextTypes,
  ),
  ...additionalOptions,
})

export function mountWrap(node) {
  return mount(node, createMixedContext({router}))
}

// export function shallowWrap(node) {
//   return shallow(node, createContext())
// }
