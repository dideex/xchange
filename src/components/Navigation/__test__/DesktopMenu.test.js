import React from 'react'
import Component from '../DesktopMenu'
// import {Provider as MobxProvider} from 'mobx-react'

import UserStore from '../../../store/User'
import {mountWrap} from '../../../helpers/router-intl-context'
import {MobxProvider} from '../../../helpers/mobx'
import {delay} from '../../../helpers'

const NO_OP = () => {}

describe('Navigation: desktop menu', () => {
  let userStore = new UserStore()
  const props = {
    handleClick: NO_OP,
    showLangMenu: false,
    showAuthMenu: false,
    closeBothMenu: NO_OP,
    toggleLangMenu: NO_OP,
    toggleAuthMenu: NO_OP,
  }

  beforeEach(() => {
    userStore = new UserStore()
  })

  describe('Markup', () => {
    it('Basic markup', () => {
      const wrapper = mountWrap(<Component {...props} />)

      expect(wrapper.html()).toMatchSnapshot()
    })

    it('Lang menu', async () => {
      const wrapper = mountWrap(
        <MobxProvider userStore={userStore}>
          <Component {...props} showLangMenu />
        </MobxProvider>,
      )

      await delay(() => {}, 500)
      expect(wrapper.html()).toMatchSnapshot()
    })
    it('Auth menu', async () => {
      const wrapper = mountWrap(
        <MobxProvider userStore={userStore}>
          <Component {...props} showAuthMenu />
        </MobxProvider>,
      )

      await delay(() => {}, 500)
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
