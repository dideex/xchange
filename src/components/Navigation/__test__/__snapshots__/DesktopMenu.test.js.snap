// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation: desktop menu Markup Auth menu 1`] = `"<nav class=\\"css-1hwc4ij\\"><a href=\\"/\\"><span>Home</span></a><a href=\\"/reservi\\"><span>Reserves</span></a><a href=\\"/o-nas\\"><span>About us</span></a><a href=\\"/faq\\"><span>FAQ</span></a><span data-testid=\\"lang-toggler\\"><span>Language</span><svg style=\\"width: 20px; margin-bottom: -2px; padding-left: 5px;\\" class=\\"\\" x=\\"0px\\" y=\\"0px\\" viewBox=\\"0 0 400 400\\"><path style=\\"fill: #aee239;\\" d=\\"M184.9,344.1L8.2,167.4c-8.5-8.5-8.5-22.3,0-30.9l20.6-20.6c8.5-8.5,22.3-8.5,30.8,0l140.7,140.1l140.7-140.1 c8.5-8.5,22.3-8.5,30.8,0l20.6,20.6c8.5,8.5,8.5,22.3,0,30.9L215.8,344.1C207.3,352.6,193.5,352.6,184.9,344.1L184.9,344.1z\\"></path></svg></span><span class=\\"auth-menu-item\\"><svg style=\\"width: 33px;\\" class=\\"\\" viewBox=\\"0 0 33 38\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path style=\\"fill: #8fbe00;\\" d=\\"M16.5 19C21.7078 19 25.9286 14.7473 25.9286 9.5C25.9286 4.25273 21.7078 0 16.5 0C11.2922 0 7.07143 4.25273 7.07143 9.5C7.07143 14.7473 11.2922 19 16.5 19ZM23.1 21.375H21.8699C20.2346 22.132 18.4152 22.5625 16.5 22.5625C14.5848 22.5625 12.7728 22.132 11.1301 21.375H9.9C4.43437 21.375 0 25.843 0 31.35V34.4375C0 36.4043 1.58371 38 3.53571 38H29.4643C31.4163 38 33 36.4043 33 34.4375V31.35C33 25.843 28.5656 21.375 23.1 21.375Z\\"></path></svg><span class=\\"css-j7t170\\"><a href=\\"/lichnii-kabinet\\"><span>Signin</span></a><a href=\\"/registracya\\"><span>Signup</span></a></span></span></nav>"`;

exports[`Navigation: desktop menu Markup Basic markup 1`] = `"<nav class=\\"css-1hwc4ij\\"><a href=\\"/\\"><span>Home</span></a><a href=\\"/reservi\\"><span>Reserves</span></a><a href=\\"/o-nas\\"><span>About us</span></a><a href=\\"/faq\\"><span>FAQ</span></a><span data-testid=\\"lang-toggler\\"><span>Language</span><svg style=\\"width: 20px; margin-bottom: -2px; padding-left: 5px;\\" class=\\"\\" x=\\"0px\\" y=\\"0px\\" viewBox=\\"0 0 400 400\\"><path style=\\"fill: #aee239;\\" d=\\"M184.9,344.1L8.2,167.4c-8.5-8.5-8.5-22.3,0-30.9l20.6-20.6c8.5-8.5,22.3-8.5,30.8,0l140.7,140.1l140.7-140.1 c8.5-8.5,22.3-8.5,30.8,0l20.6,20.6c8.5,8.5,8.5,22.3,0,30.9L215.8,344.1C207.3,352.6,193.5,352.6,184.9,344.1L184.9,344.1z\\"></path></svg></span><span class=\\"auth-menu-item\\"><svg style=\\"width: 33px;\\" class=\\"\\" viewBox=\\"0 0 33 38\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path style=\\"fill: #8fbe00;\\" d=\\"M16.5 19C21.7078 19 25.9286 14.7473 25.9286 9.5C25.9286 4.25273 21.7078 0 16.5 0C11.2922 0 7.07143 4.25273 7.07143 9.5C7.07143 14.7473 11.2922 19 16.5 19ZM23.1 21.375H21.8699C20.2346 22.132 18.4152 22.5625 16.5 22.5625C14.5848 22.5625 12.7728 22.132 11.1301 21.375H9.9C4.43437 21.375 0 25.843 0 31.35V34.4375C0 36.4043 1.58371 38 3.53571 38H29.4643C31.4163 38 33 36.4043 33 34.4375V31.35C33 25.843 28.5656 21.375 23.1 21.375Z\\"></path></svg></span></nav>"`;

exports[`Navigation: desktop menu Markup Lang menu 1`] = `"<nav class=\\"css-1hwc4ij\\"><a href=\\"/\\"><span>Home</span></a><a href=\\"/reservi\\"><span>Reserves</span></a><a href=\\"/o-nas\\"><span>About us</span></a><a href=\\"/faq\\"><span>FAQ</span></a><span data-testid=\\"lang-toggler\\"><span>Language</span><svg style=\\"width: 20px; margin-bottom: -2px; padding-left: 5px;\\" class=\\"\\" x=\\"0px\\" y=\\"0px\\" viewBox=\\"0 0 400 400\\"><path style=\\"fill: #aee239;\\" d=\\"M184.9,344.1L8.2,167.4c-8.5-8.5-8.5-22.3,0-30.9l20.6-20.6c8.5-8.5,22.3-8.5,30.8,0l140.7,140.1l140.7-140.1 c8.5-8.5,22.3-8.5,30.8,0l20.6,20.6c8.5,8.5,8.5,22.3,0,30.9L215.8,344.1C207.3,352.6,193.5,352.6,184.9,344.1L184.9,344.1z\\"></path></svg><span class=\\"css-j7t170\\"><span data-testid=\\"ru-RU\\">Рус</span><span data-testid=\\"en-US\\">Eng</span><span data-testid=\\"fr-FR\\">Fr</span><span data-testid=\\"zh-ZH\\">Cn</span></span></span><span class=\\"auth-menu-item\\"><svg style=\\"width: 33px;\\" class=\\"\\" viewBox=\\"0 0 33 38\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path style=\\"fill: #8fbe00;\\" d=\\"M16.5 19C21.7078 19 25.9286 14.7473 25.9286 9.5C25.9286 4.25273 21.7078 0 16.5 0C11.2922 0 7.07143 4.25273 7.07143 9.5C7.07143 14.7473 11.2922 19 16.5 19ZM23.1 21.375H21.8699C20.2346 22.132 18.4152 22.5625 16.5 22.5625C14.5848 22.5625 12.7728 22.132 11.1301 21.375H9.9C4.43437 21.375 0 25.843 0 31.35V34.4375C0 36.4043 1.58371 38 3.53571 38H29.4643C31.4163 38 33 36.4043 33 34.4375V31.35C33 25.843 28.5656 21.375 23.1 21.375Z\\"></path></svg></span></nav>"`;
